src/Controller/Api/GalleryController.php

```diff
@@ -32,11 +31,14 @@
 use Symfony\Component\Form\FormInterface;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\Annotation\Route;
 
 /**
  * @final since sonata-project/media-bundle 3.21.0
  *
  * @author Hugo Briand <briand@ekino.com>
+ *
+ * @Route(defaults={"_format": "json"}, requirements={"_format": "json|xml|html"})
```

Comment:
Can this routes be defined on an xml? otherwise this means it is mandatory to use the sensio extra framework bundle in order to use the api.

Reply:
`sensio/framework-extra-bundle` is also require for `view_response_listener`.
https://github.com/FriendsOfSymfony/FOSRestBundle/blob/3.x/composer.json#L66

Reply:
AFAIK, this is not using "sensio/framework-extra-bundle", since this annotation is `Symfony\Component\Routing\Annotation\Route`, not `Sensio\Bundle\FrameworkExtraBundle\Configuration\Route`.

Reply:
Not sure where it comes there requirement, but this PR added it on require-dev, if it is not required, please remove it. Otherwise, if it is required to the api to work, it should be on docs (on the install doc, near the other two bundles required: nelmio an rest)

Reply:
The annotation `@Symfony\Component\Routing\Annotation\Route` is provided by "symfony/routing", which is already a requirement (in the `require` definition of `composer.json`): https://github.com/sonata-project/SonataMediaBundle/blob/55d4a34311b919264fdfd05a39b4a3aa5caed27d/composer.json#L52.

Reply:
I check it. It is require in `--prefer-lowest`. In others builds I can remove it. So another dump time :smile: .

Reply:
It seems an insane conversation. If we are talking about `@Symfony\Component\Routing\Annotation\Route`, there is nothing to do with `sensio/framework-extra-bundle`. Could you please confirm if you agree?

Reply:
Error: Class 'Sensio\Bundle\FrameworkExtraBundle\Configuration\Template' not found  :smile:

Reply:
I know it is coming from symfony/routing.. but if you follow the docs on Symfony it is explained. https://symfony.com/doc/current/routing.html#creating-routes-as-annotations

Reply:
That `composer req annotation` installs de sensio extra bundle

Reply:
The only thing to know here is: is this the only way to do this thing? if yes, it would be nice to document the sensio/extra-bundle near the nelmio an rest dependencies.

If not, it would be nice to use a way where you don't need to add another dependency (or atleast IMO reducing dependencies is always great).

Reply:
@jordisala1991 It is require only in `--prefer-lowest`, so it is possible to bump some package and remove it. I will try to do it today.

Reply:
>If not, it would be nice to use a way where you don't need to add another dependency (or atleast IMO reducing dependencies is always great).

I think we could use other format.